cpp/src/join/semi_join.cu

```diff
@@ -91,22 +86,35 @@ std::unique_ptr<cudf::table> left_semi_anti_join(
     return empty_like(left.select(return_columns));
   }
 
-  if ((join_kind::LEFT_ANTI_JOIN == JoinKind) && (0 == right.num_rows())) {
+  auto const left_num_rows  = left.num_rows();
+  auto const right_num_rows = right.num_rows();
```

Comment:
Are these necessary?

Reply:
Considering there are now 3 columns: `left`, `left_selected`, `left_updated` I thought it would be clearer to set/use a variable then to randomly pick one of the 3 columns. Same for `right_num_rows`.